### PR TITLE
Add a new http3 test server that always rejects zero rtt

### PR DESCRIPTION
**Pull request by KershawChang**
This patch simply adds `disallow_zerortt_server` to facilitate 0-Rtt reject test.

### CODE REVIEW

File: neqo-transport/src/connection/tests/mod.rs [hunk 1]
```diff
@@ -77,6 +77,22 @@ pub fn default_server() -> Connection {
     c
 }
 
+pub fn disallow_zerortt_server() -> Connection {
```
Comment (martinthomson) [inline neqo-transport/src/connection/tests/mod.rs:80]: One-off things like this don't need functions to be added to the base fixture.  Just include this stuff inline.

File: neqo-transport/src/connection/tests/zerortt.rs [hunk 2]
```diff
@@ -129,22 +125,8 @@ fn zero_rtt_send_reject() {
     client
         .enable_resumption(now(), token)
         .expect("should set token");
-    let mut server = Connection::new_server(
```
Comment (martinthomson) [inline neqo-transport/src/connection/tests/zerortt.rs:132]: This is a different test case.  I would prefer that you add a new test case for your new thing.

File: neqo-crypto/src/agent.rs [hunk 3]
```diff
@@ -867,6 +867,15 @@ impl ZeroRttChecker for AllowZeroRtt {
     }
 }
 
+#[derive(Debug)]
+pub struct DisallowZeroRtt {}
+impl ZeroRttChecker for DisallowZeroRtt {
```
Comment (martinthomson) [inline neqo-crypto/src/agent.rs:872]: This doesn't need to be public.  There is no situation, except for testing, where you would enable 0-RTT but then reject it.

**Review by martinthomson — CHANGES_REQUESTED** [review]
If you are looking to implement rejection of 0-RTT for neqo-server, then the implementation of the rejection code should be local to that file.

**Comment by ddragana** [comment]
This has bin resolved by #1150
